subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemBuildLifecycleListener.java

```diff
@@ -34,7 +34,7 @@ public VirtualFileSystemBuildLifecycleListener(WatchingAwareVirtualFileSystem vi
 
     @Override
     public void afterStart(GradleInternal gradle) {
-        StartParameter startParameter = gradle.getStartParameter();
+        StartParameterInternal startParameter = (StartParameterInternal) gradle.getStartParameter();
```

Comment:
`GradleInternal` should have an override to return a `StartParameterInternal`.

Reply:
If `GradleInternal` has the override, does that mean that Groovy then will compile against `StartParameterInternal`?

I can add the override. I vaguely remember that had some bad side-effects, though probably in a different context.

Reply:
I think what you are saying is not a problem here. Perhaps you are thinking about the bug where the static compiler would pick a super-interface to compile against at random. That bug was fixed in 2.4.11, and it wouldn't apply here anyway.

Reply:
This is the bug: https://issues.apache.org/jira/browse/GROOVY-8142